server/pages/index.js

```diff
@@ -474,13 +480,46 @@ function render404( request, response ) {
 	response.status( 404 ).send( renderJsx( '404', ctx ) );
 }
 
+function handleLocaleSubdomains( req, res, next ) {
+	const langRouteParams = config( 'magnificent_non_en_locales' ).join( '|' );
+	const hostnameGroup = `\\.${ config( 'hostname' ).replace( /\./g, '\\.' ) }`;
+	const langSubDomainRegEx = new RegExp( `^(${ langRouteParams })(${ hostnameGroup })` );
+	const langSubDomainMatch = req.hostname.match( langSubDomainRegEx );
```

Comment:
This is probably a matter of personal taste, but I find that amount of Regex-fu a bit confusing and cumbersome (esp the escaping of the periods). What about adding an `endsWith( req.hostname, config( 'hostname' ))` check, and if it returns `true`, get the substring left of it (trimming any trailing periods), and checking whether `includes( config( 'magnificent_non_en_locales' ), thatSubstring )`?